readthedocs/api/v2/views/integrations.py

```diff
@@ -459,6 +464,16 @@ def handle_webhook(self):
             except KeyError:
                 raise ParseError('Parameter "ref" is required')
 
+        # Re-sync repositories for the user if any permission has changed
+        if event == GITHUB_MEMBER:
+            uid = self.data.get('member').get('id')
+            socialaccount = SocialAccount.objects.get(
+                provider=GitHubProvider.id,
+                uid=uid,
+            )
+            service = GitHubService(user=socialaccount.user, account=socialaccount)
+            service.sync()
```

Comment:
We may want to do this async via a Celery task. We already have [`sync_remote_repositories`](https://github.com/readthedocs/readthedocs.org/blob/2565e275ff1c3d45af1d3b9e387a7bc5591d8c41/readthedocs/oauth/tasks.py#L27) but we should add an argument to accept which service to sync instead of doing it on all of them --if we want to be precise here.

Reply:
Yeah, :+1: on moving this to a celery task. Lately, we have definitely been feeling the effects of executing these sorts of tasks in-process on the webs. `sync()` is perfectly fine otherwise :+1: